api/session_test.go

```diff
@@ -21,6 +21,13 @@ var _ = Describe("Session", func() {
 		session = &Session{bus}
 	})
 
+	Describe("#New", func() {
```

Comment:
Okay, how about you submit the PR without the integration test for `JoinPage`, and I'll add the integration test sometime after I merge it.

As for `GetSessionURL()`, I'd still like you to leave it out. You should have the session URL already, because you had to pass it to `JoinPage`.